CRM/CiviGeometry/BAO/Geometry.php

```diff
@@ -434,37 +434,167 @@ public static function wkt2kml($wkt) {
+      'precheck_relationships' => true,
+    ];
+    $params = $params + $defaultParams;
+
+    // Step 1. Find all the candidate addresses within the bounds of the geometry Use a table to
+    // store a potentially very large number of addresses
+    $bounds = civicrm_api3('Geometry', 'getbounds', ['id' => $geometry_id])['values'][$geometry_id];
+    $tmpTbl = CRM_Utils_SQL_TempTable::build();
+    if ($params['keep_temp_table']) {
+      $tmpTbl->setDurable();
+    }
+    else {
+      $tmpTbl->setAutodrop(true);
+    }
+    $tmpTbl->createWithColumns('address_id INT');
+
+    $tempTableName = $tmpTbl->getName();
+    if ($params['precheck_relationships']) {
+      CRM_Core_DAO::executeQuery("
+        INSERT INTO $tempTableName (address_id)
+        SELECT ca.id
+        FROM
+          civicrm_address ca
+        WHERE
+              ca.geo_code_2 >= %1
+          AND ca.geo_code_2 <= %2
+          AND ca.geo_code_1 <= %3
+          AND ca.geo_code_1 >= %4
+          AND NOT EXISTS (
+            SELECT *
+            FROM civigeometry_geometry_entity cge
+            WHERE
+              cge.entity_id = ca.id
+              AND cge.geometry_id = %5
+              AND cge.entity_table = 'civicrm_address'
+          )
+        ORDER BY ca.id
+      ", [
+        1 => [$bounds['left_bound'], 'Float'],
+        2 => [$bounds['right_bound'], 'Float'],
+        3 => [$bounds['top_bound'], 'Float'],
+        4 => [$bounds['bottom_bound'], 'Float'],
+        5 => [$geometry_id, 'Integer']
+      ]);
     }
     else {
-      return [];
+      CRM_Core_DAO::executeQuery("
+        INSERT INTO $tempTableName (address_id)
+        SELECT ca.id
+        FROM civicrm_address ca
+        WHERE
+              ca.geo_code_2 >= %1
+          AND ca.geo_code_2 <= %2
+          AND ca.geo_code_1 <= %3
+          AND ca.geo_code_1 >= %4
+        ORDER BY ca.id
+      ", [
+        1 => [$bounds['left_bound'], 'Float'],
+        2 => [$bounds['right_bound'], 'Float'],
+        3 => [$bounds['top_bound'], 'Float'],
+        4 => [$bounds['bottom_bound'], 'Float'],
+      ]);
+    }
+
+    $numCandidates = CRM_Core_DAO::singleValueQuery("SELECT COUNT(*) FROM $tempTableName");
+
+    $numBatches = ceil($numCandidates / $params['batch_size']);
+    $offset = 0;
+    for ($batchN=1; $batchN <= $numBatches; $batchN++) {
+      $candidates = self::getAddressesFetchCandidateBatch($tempTableName, $offset, $params['batch_size']);
+
+      if (count($candidates)) {
+        foreach ($candidates as $candidate) {
+          if (self::getAddressesGeometryContainsCandidate($geometry_id, $candidate['address_id'])) {
+            yield ['geometry_id' => $geometry_id, 'address_id' => $candidate['address_id']];
+          }
+        }
+
+        $offset += $params['batch_size'];
+      }
+      else {
+        throw new CRM_Core_ExecptionException(E::ts("No candidate addresses, but batches still remaining to process. That doesn't make any sense."));
```

Comment:
@unfinishedteleporter too many Execptions here if you know what I mean

Reply:
Inxception